### PR TITLE
rgw_sal_motr: [CORTX-34113]Copy-Object API :Handled UnknownError while performing copy object using tagging directive COPY 

### PR DESCRIPTION
**Pull request by shraddhaghatol**
Problem: Getting "UnknownError" while performing copy object using tagging directive COPY on source object with no tags
Solution : Error occurs only when we put fresh object (without tagging) in bucket and perform copy-object operation with --tagging -directive COPY. No need to return error if empty tags are present. Even if tags are empty copy operation should get successful with --tagging-directive COPY.

If tags are empty we were returning with -66 error from copy_object_same_zone() function.

Signed-off-by: shraddhaghatol <shraddha.j.ghatol@seagate.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by shriya-deshmukh — APPROVED** [review]

File: src/rgw/rgw_sal_motr.cc [hunk 1]
```diff
@@ -2489,10 +2489,6 @@ int MotrObject::copy_object_same_zone(RGWObjectCtx& obj_ctx,
   if (tagging_drctv) {
     if (strcasecmp(tagging_drctv, "COPY") == 0) {
       rc = read_op->get_attr(dpp, RGW_ATTR_TAGS, tags_bl, y);
-      if (rc < 0) {
-        ldpp_dout(dpp, LOG_ERROR) <<__func__ << ": ERROR: read op for object tags failed rc=" << rc << dendl;
```
Comment (cdeshmukh) [inline src/rgw/rgw_sal_motr.cc:2493]: keep log at debug level stating no tags seen for this src object

Comment (shraddhaghatol) [reply src/rgw/rgw_sal_motr.cc:2493]: Addressed.

**Review by cdeshmukh — APPROVED** [review]

**Comment by shraddhaghatol** [comment]
Ran ceph tests. All tests are successful.
following is the job link::
https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/S3Server/job/S3-Compatibility-Test-Ceph-RGW/240/
image

**Review by sachitanands — APPROVED** [review]
looks good

**Pull request merged**